OpenRA.Mods.Common/Traits/Player/TimeLimitManager.cs

```diff
@@ -0,0 +1,133 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2019 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("This trait allows setting a time limit on matches. Attach this to the Player actor.")]
+	public class TimeLimitManagerInfo : ITraitInfo, ILobbyOptions
+	{
+		[Desc("Label that will be shown for the time limit option in the lobby.")]
+		public readonly string TimeLimitLabel = "Time Limit";
+
+		[Desc("Tooltip description that will be shown for the time limit option in the lobby.")]
+		public readonly string TimeLimitDescription = "Time Limit before a match ends automatically.";
+
+		[Desc("Time Limit options that will be shown in the lobby dropdown. Values are in minutes.")]
+		public readonly int[] TimeLimitOptions = { 0, 20, 30, 40, 60, 90 };
```

Comment:
Can we add 10 minutes too?

Reply:
Sure, but don't you think that 10 minutes is a bit short (considering the insights from http://www.openra.net/news/2017-01-stats-post/)?  Maybe 15 minutes would be more reasonable?

Reply:
I like the idea of keeping one option slightly less than what we'd consider reasonable, to help prevent the inevitable complaints from outliers.